jth-tests/src/test/java/com/redhat/jenkins/nodesharing/PoolTest.java

```diff
@@ -132,6 +138,57 @@ public void readConfigFromRepo() {
         assertFalse(Pool.ADMIN_MONITOR.getErrors().toString(), Pool.ADMIN_MONITOR.isActivated());
     }
 
+    @Test
+    public void getSnapshotFailedRemote() throws Exception {
+        final File workingDir = new File(Jenkins.getInstance().getRootDir() + "/node-sharing/config");
+        final ConfigRepo repo = new ConfigRepo(Pool.getInstance().getConfigRepoUrl(), workingDir);
+
+        // Case 1 - recovered from previously stored config repo
+        ConfigRepo mockConfigRepo = spy(repo);
+        mockConfigRepo.getSnapshot();
+        when(mockConfigRepo.getRemoteHead(Mockito.any(TaskLog.class))).thenThrow(new GitException("Not available"));
+        boolean exceptionThrown = false;
+        try {
+            mockConfigRepo.getSnapshot();
+        } catch (TaskLog.TaskFailed e) {
+            exceptionThrown = true;
+        }
+        assertFalse("TaskLog.TaskFailed exceptions shouldn't be thrown", exceptionThrown);
+
+        // Case 2 - trying to recover from not existing config repo (wasn't saved previously)
+        mockConfigRepo = spy(repo);
+        when(mockConfigRepo.getRemoteHead(Mockito.any(TaskLog.class))).thenThrow(new GitException("Git not available - intentionally"));
+        exceptionThrown = false;
+        try {
+            mockConfigRepo.getSnapshot();
+        } catch (TaskLog.TaskFailed e) {
+            exceptionThrown = true;
+            assertThat(e, instanceOf(TaskLog.TaskFailed.class));
+            assertThat(e.getMessage(), containsString("Unable to read snapshot from "));
+            assertThat(e.getLog().readContent(), containsString("Git not available - intentionally"));
+        }
+        assertTrue("TaskLog.TaskFailed exceptions should be thrown", exceptionThrown);
```

Comment:
Conventionally, this is done by `Assert#fail(String)` at the end of the try block.